src/index/hgraph_zparameters.h

```diff
@@ -0,0 +1,69 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <nlohmann/json.hpp>
+#include <string>
+
+#include "index_common_param.h"
+
+namespace vsag {
+class HgraphParameters {
+public:
+    explicit HgraphParameters(const IndexCommonParam& common_param, const std::string& str = "");
+
+    void
+    ParseStringParam(const std::string& str);
+
+    void
+    CheckAndSetKeyValue(const std::string& key, nlohmann::json& value);
+
+    std::string
+    GetString() {
+        this->sync_string_by_json();
+        return this->str_;
+    }
+
+    nlohmann::json
+    GetJson() {
+        this->sync_json_by_string();
+        return this->json_;
+    }
+
+private:
+    void
+    sync_json_by_string() {
```

Comment:
refresh_json ?

Reply:
I think the refresh_json_by_string is better

Reply:
OK